### DISCUSSION TITLE
I can't get the expected error information with using `analysis.diagnostic()` in rust_analyzer api.

### DISCUSSION THREAD

**Discussion by baiguoname** [rust-lang/rust-analyzer#15599] [Q&A] [answered]
Suppose I have the source file, which has the following code:
```
fn main() {
    1usize.undefined_method();
}
```
then the method `analysis.diagnostic` will return the error `usize has no method undefined_method`.
But if I have the source file with following code:
```
fn main() {
  let a = undefined_function();
}
```
then the method `analysis.diagnostic` will not return error.
The `DiagnosticsConfig` is set to be `DiagnosticsConfig::test_sample()`.
What am I doing wrong?

**Comment by Veykril** [comment]
The reason for that is that we haven't implemented a diagnostic for unresolved paths yet.

**Comment by baiguoname** [comment]
> The reason for that is that we haven't implemented a diagnostic for unresolved paths yet.

If I use the latter code in vscode, there will be an indication of error. Does this indication not come from diagnostic?

**Reply by Veykril** [reply, marked as answer]
That one probably comes from us invoking `cargo check` on save which is just the rust compiler's diagnostic being shown. The diagnostics function you are calling here only contains the rust-analyzer native diagnostics.
